src/utils/extensionUtil.ts

```diff
@@ -3,26 +3,23 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 import * as vscode from 'vscode';
-import { DialogResponses, IActionContext } from '@microsoft/vscode-azext-utils';
-import { localize } from '../localize';
-import { openUrl } from '../utils/openUrl';
+import * as fs from 'fs';
+import * as path from 'path';
 
-// tslint:disable-next-line:export-name
-export function checkCsharpExtensionInstalled(actionContext: IActionContext): void {
-    const csharpExtension = vscode.extensions.getExtension("ms-dotnettools.csharp") || vscode.extensions.getExtension("ms-vscode.csharp");
-    if (!csharpExtension) {
-        const message: string = localize('csharpExtensionNotInstalled', 'You must have the VSCode CSharp extension installed to improve policy authoring experience.');
+export async function showReleaseNotes(context: vscode.ExtensionContext): Promise<void> {
+    const packageJson = JSON.parse(fs.readFileSync(path.join(context.extensionPath, 'package.json'), 'utf8'));
+    const currentVersion = packageJson.version;
+    const lastVersion = context.globalState.get<string>('lastShownVersion');
 
-        if (!actionContext.errorHandling.suppressDisplay) {
-            // don't wait
-            vscode.window.showErrorMessage(message, DialogResponses.learnMore).then(async (result) => {
-                if (result === DialogResponses.learnMore) {
-                    await openUrl('https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp');
-                }
-            });
-            actionContext.errorHandling.suppressDisplay = true;
-        }
+    // Only show if version has changed and not shown before
+    if (lastVersion !== currentVersion) {
```

Comment:
need to check lastVersion exist

Reply:
Current version in marketplace does not set lastShownVersion, so we also need to pop up release note if lastVersion is undefined.

Reply:
so if the lastVersion is undefined, this sentence will crash?